### ISSUE TITLE
Global costmap resized by static layer even if layered_costmap_->isSizeLocked() is set to true

### ISSUE THREAD

**Issue by ghost** [ros-planning/navigation#768]
In the function staticLayer::incomingMap(navigation/costmap_2d/plugins/static_layer.cpp line 166), the first if statement results in master map(layered_costmap) being resized even if the setting for layered_costmap_->isSizeLocked is set to true. This seems unintended to me and I was wondering the changes below should be made. 

There are few other changes to be made for this to fully work, but for now I just wanted to ask whether I was correct in reading this as a bug.

The lines in question: 
https://github.com/ros-planning/navigation/blob/7fa21a2759806ffea7fee9be436ab27b6b3c102f/costmap_2d/plugins/static_layer.cpp#L174-L185

And the changes I'm suggesting:
```
-  if (!layered_costmap_->isRolling() && (master->getSizeInCellsX() != size_x ||
+  if (!layered_costmap_->isRolling() && !layered_costmap_->isSizeLocked()
+   && (master->getSizeInCellsX() != size_x ||
    master->getSizeInCellsY() != size_y ||
    master->getResolution() != new_map->info.resolution ||
    master->getOriginX() != new_map->info.origin.position.x ||
-   master->getOriginY() != new_map->info.origin.position.y ||
-   !layered_costmap_->isSizeLocked()))
+   master->getOriginY() != new_map->info.origin.position.y)) 
```

**Comment by moriarty** [comment]
Sorry this comment won't answer your question. I just saw a wall of unformatted text and had to say something.

I guess there is no "how to file a bug" for this repository, but some more strict repos would require a little bit more basic information for [example](https://github.com/ev3dev/ev3dev/blame/ev3dev-jessie/ISSUE_TEMPLATE.md)

FYI, if you find the lines in GitHub, you can click on them, and then select "copy permalink" which is much easier to read... Then you can also use three ` marks, to type code blocks and use diff format to show clearer what you mean.

https://github.com/ros-planning/navigation/blob/7fa21a2759806ffea7fee9be436ab27b6b3c102f/costmap_2d/plugins/static_layer.cpp#L174-L185

I _might_ have made a typo, but from reading your text and typing it again (this is easier if you try it on your machine first and just copy some of the output of git diff), I think you want to replace the lines above, with:

```diff
-  if (!layered_costmap_->isRolling() && (master->getSizeInCellsX() != size_x ||
+  if (!layered_costmap_->isRolling() && !layered_costmap_->isSizeLocked()
+   && (master->getSizeInCellsX() != size_x ||
    master->getSizeInCellsY() != size_y ||
    master->getResolution() != new_map->info.resolution ||
    master->getOriginX() != new_map->info.origin.position.x ||
-   master->getOriginY() != new_map->info.origin.position.y ||
-   !layered_costmap_->isSizeLocked()))
+   master->getOriginY() != new_map->info.origin.position.y))
```

A quick look at git-blame view from within GitHub shows that these lines were last modified with PR https://github.com/ros-planning/navigation/pull/324 which was @mikeferguson & @reinzor

**Comment by ghost** [comment]
Thanks for the suggestions! Will update and reopen with corrections @moriarty

**Comment by moriarty** [comment]
@jinder1s I didn’t mean be rude or scare you away, I just wanted to help pin point the issue you were trying to report

**Comment by ghost** [comment]
@moriarty  no no, I genuinely appreciate the advice. Its just the confluence of many events yesterday did not allow me to edit this issue request till now.

**Comment by mikeferguson** [comment]
@jinder1s that does seem like an error, and you appear to have the correct fix -- please open a PR

**Comment by DLu** [comment]
Closed via #792
